src/mergeSort.js

```diff
@@ -0,0 +1,35 @@
+//time complexity: O(nlogn)
+
+export default function mergeSort(arr) {
+  if(arr.length < 2){
+    return arr
+  }
+
+  let middle = parseInt(arr.length/2)
+  const left = arr.splice(0, middle)
+  const right = arr.splice(middle, arr.length)
+
+  return merge(mergeSort(left), mergeSort(right))
```

Comment:
I like the idea of using a recursive function here, but the end result is an array of two values. I think you could refactor it as maybe one large array of arrays that you can compare in the merge function below. I tried a couple of different options without changing your structure and could not get it to return the behavior I wanted. You also may have a problem with an odd number of values, since you would be calling merge like this: merge( [array] ). This could also affect behavior as you try to compare values to undefined in the merge function. I wish I could say more, but I have not done these algorithms yet.